service/scaladsl/broker/src/main/scala/com/lightbend/lagom/scaladsl/broker/TopicProducerCommand.scala

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.scaladsl.broker
+
+import akka.persistence.query.Offset
+
+import scala.collection.immutable
+
+sealed trait TopicProducerCommand[Message]
+
+object TopicProducerCommand {
+  case class EmitMultipleAndCommit[Message](messages: immutable.Seq[Message], offset: Offset)
+      extends TopicProducerCommand[Message]
+  case class EmitAndCommit[Message](message: Message, offset: Offset) extends TopicProducerCommand[Message]
+  case class Commit[Message](offset: Offset)                          extends TopicProducerCommand[Message]
+}
```

Comment:
Using case classes will difficult future evolution of the API. 

This should be marked as `@ApiMayChange` or refactorred as non-case class.

Reply:
Especially considering the upcoming need for metadata (#2835 ).

Reply:
Changed to normal class.